### PR TITLE
Iz close dropdown fix GAWB-2121

### PR DESCRIPTION
**Pull request by zarsky-broad**

### CODE REVIEW

**Review by MattPutnam — COMMENTED** [review]
👍 Looks good.  Congrats on a legit usage of a moderately obscure and highly situational `let` form.

**Comment by zarsky-broad** [comment]
I'm vaguely concerned by the fact that the test seems to fail on this, though. Thoughts @dmohs?

File: src/cljs/main/broadfcui/common.cljs [hunk 1]
```diff
@@ -297,30 +297,36 @@
...
-                  (if (:close-on-click props)
-                    (.on element$ "click.zf.dropdown"
-                         (fn [_]
-                           (js/setTimeout                   ; allow click handlers to fire
-                            #(.foundation element$ "close")
-                            0)))
+           (letfn [(clean-up []
+                     (.off (js/$ "body") "click.zf.dropdown" body-fn)
+                     (.off element$ "click.zf.dropdown" element-fn))
+                   (element-fn [_]
```
Comment (dmohs) [inline src/cljs/main/broadfcui/common.cljs]: `close-on-element-click`?

File: src/cljs/main/broadfcui/common.cljs [hunk 2]
```diff
@@ -297,30 +297,36 @@
...
+                     (js/setTimeout                         ; allow click handlers to fire
+                      #(.foundation element$ "close")
+                      0)
```
Comment (dmohs) [inline src/cljs/main/broadfcui/common.cljs]: Is the extra vertical space usage here worth it?

File: src/cljs/main/broadfcui/common.cljs [hunk 3]
```diff
@@ -297,30 +297,36 @@
...
+                     (clean-up))
+                   (body-fn [e]
```
Comment (dmohs) [inline src/cljs/main/broadfcui/common.cljs]: `close-on-body-click`?

File: src/cljs/main/broadfcui/common.cljs [hunk 4]
```diff
@@ -297,30 +297,36 @@
...
+                    (swap! state dissoc :render-contents?)
+                    (after-update #(this :-render-dropdown))))
+             (.on button$
+                  "click"
+                  (fn [_]
+                    (swap! state assoc :render-contents? true)
+                    (after-update #(this :-render-dropdown))
+                    (when (:close-on-click props)
+                      (.on element$ "click.zf.dropdown"
+                           element-fn))
```
Comment (dmohs) [inline src/cljs/main/broadfcui/common.cljs]: `(.on element$ "click.zf.dropdown" close-on-element-click)`

File: src/cljs/main/broadfcui/common.cljs [hunk 5]
```diff
@@ -297,30 +297,36 @@
...
                     (.on (js/$ "body")
                          "click.zf.dropdown"
```
Comment (dmohs) [inline src/cljs/main/broadfcui/common.cljs]: `(.on (js/$ "body") "click.zf.dropdown" close-on-body-click)`

**Review by dmohs — APPROVED** [review]
I think a smidge more vertical space reduction will make this block appears less daunting.

**Pull request merged**